pyproject.toml

```diff
@@ -16,7 +16,9 @@ exclude = ["tests", "tests.*"]
 
 [tool.poetry.dependencies]
 python = "3.9.*"
-"discord.py" = {url = "https://github.com/Rapptz/discord.py/archive/beafaa8a8b8006357c2a1ca6802ed80e000f4cda.zip"}
+"discord.py" = {url = "https://github.com/Rapptz/discord.py/archive/414759f3723a3fe632ecfc1343a4164a51cd2940.zip"}
+async-rediscache = { version = "0.2.0", extras = ["fakeredis"] }
+statsd = "3.3.0"
```

Comment:
Do you think it would be better to make these extra dependencies? Support for redis/statsd could be disabled if these aren't installed.

Reply:
I've attempted to do this here https://github.com/python-discord/bot-core/pull/42/commits/a904fcb76d97af7076755e3c6f0e29f2b5d7642e. This is my first time doing something like this so would appreciate feedback on whether there is a better way.

Reply:
I think the approach is _okay_. I did have some minor concerns, but I am not sure of a better way.

Reply:
I reverted this and only made async-redis cache optional. The solution I had for stats was hacky and messed with pylance enough that it couldn't pick up that it had parent class methods.